src/java/org/apache/cassandra/db/ColumnFamilyStore.java

```diff
@@ -1839,13 +1843,13 @@ public ClusteringComparator getComparator()
 
     public TableSnapshot snapshotWithoutFlush(String snapshotName)
```

Comment:
There is another usage of this method in `ConpactionTask`:
```
if (DatabaseDescriptor.isSnapshotBeforeCompaction())
    cfs.snapshotWithoutFlush(System.currentTimeMillis() + "-compact-" + cfs.name);
```

Looks like it has a timestamp in the name as well. Should we fix this one too? If yes, probably, I'd just change the signature of this method and add `timestamp` argument.

Reply:
I introduced the method you suggested but I have also left there timestamp-less version of it which just takes snapshotName and which calls the new method with generated timestamp.